master/pkg/searcher/custom_search_test.go

```diff
@@ -0,0 +1,91 @@
+//nolint:exhaustivestruct
+package searcher
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/nprand"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+// Testing a few methods (not all because they are similar)
+// from customSearcherMethod to ensure that the methods
+// and the queue is working well.
+func TestCustomSearchMethod(t *testing.T) {
+	config := expconf.SearcherConfig{
+		RawCustomConfig: &expconf.CustomConfig{RawMaxLength: ptrs.Ptr(expconf.NewLengthInBatches(500))},
+	}
+
+	customSearchMethod := NewSearchMethod(config)
+	rand := nprand.New(0)
+	ctx := context{rand: rand}
+
+	var queue *SearcherEventQueue
+	if csMethod, ok := customSearchMethod.(CustomSearchMethod); ok {
+		queue = csMethod.getSearcherEventQueue()
+	}
```

Comment:
You can just do `queue := customSearchMethod.(CustomSearchMethod).getSearcherEventQueue()`, since you know this will work (and if it somehow didn't, we would want it to fail immediately anyway).